<launch>
    <arg name="simulation"/>

    <node pkg="face_following" type="nearest_face_following_node.py" name="nearest_face_following_node">
        <param name="simulation" value="$(arg simulation)"/>
        <param name="control_frequency" value="30"/>
        <param name="torso_control_alpha" value="0.2"/>
        <param name="head_control_p_gain" value="0.175"/>
        <param name="head_enabled" value="true"/>
        <param name="min_head_pitch_rad" value="-0.35"/>
        <param name="max_head_pitch_rad" value="0.35"/>
        <param name="nose_confidence_threshold" value="0.4"/>

        <remap from="pose/filter_state" to="nearest_face_following/filter_state"/>
    </node>

    <node pkg="face_following" type="specific_face_following_node.py" name="specific_face_following_node">
        <param name="simulation" value="$(arg simulation)"/>
        <param name="control_frequency" value="30"/>
        <param name="torso_control_alpha" value="0.2"/>
        <param name="head_control_p_gain" value="0.175"/>
        <param name="head_enabled" value="true"/>
        <param name="min_head_pitch_rad" value="-0.35"/>
        <param name="max_head_pitch_rad" value="0.35"/>
        <param name="direction_frame_id" value="odas"/>

        <remap from="target_name" to="face_following/target_name"/>
        <remap from="pose/filter_state" to="specific_face_following/filter_state"/>
    </node>

</launch>
